File: src/dotnet/FaxCore.ApiSamples/Samples/User/Message/UpdateMessageSubject.cs
using System;
using System.Threading.Tasks;

namespace FaxCore.ApiSamples.Samples.User.Message
{
    internal static class UpdateMessageSubject
    {
        public static Task<string> RunAsync(FaxCoreClient client, SampleConfig config)
        {
            if (!string.Equals(Environment.GetEnvironmentVariable("FAXCORE_CONFIRM_DESTRUCTIVE"), "true", StringComparison.OrdinalIgnoreCase))
            {
                throw new InvalidOperationException("Set FAXCORE_CONFIRM_DESTRUCTIVE=true to update a message subject.");
            }

            if (string.IsNullOrWhiteSpace(config.MessageId) || string.IsNullOrWhiteSpace(config.MessageSubject))
            {
                throw new InvalidOperationException("Set FAXCORE_MESSAGE_ID and FAXCORE_MESSAGE_SUBJECT before running user.message.subject.update.");
            }

            return client.PutAsync("/api/message/subject", new
            {
                messageId = config.MessageId,
                value = config.MessageSubject
            });
        }
    }
}

File: src/javascript/user/message/update-message-subject.mjs
import { FaxCoreClient, printResult } from '../../common/faxcoreClient.mjs';

if (process.env.FAXCORE_CONFIRM_DESTRUCTIVE !== 'true') {
  throw new Error('Set FAXCORE_CONFIRM_DESTRUCTIVE=true to update a message subject.');
}

const messageID = process.env.FAXCORE_MESSAGE_ID;
const subject = process.env.FAXCORE_MESSAGE_SUBJECT;

if (!messageID || !subject) {
  throw new Error('FAXCORE_MESSAGE_ID and FAXCORE_MESSAGE_SUBJECT are required.');
}

const client = new FaxCoreClient();
const result = await client.put('/api/message/subject', {
  messageId: messageID,
  value: subject
});

printResult('Update message subject', result);
